src/openzeppelin/tests/upgrades/test_upgradeable.cairo

```diff
@@ -0,0 +1,175 @@
+use array::ArrayTrait;
+use option::OptionTrait;
+use starknet::class_hash::ClassHash;
+use starknet::class_hash::class_hash_const;
+use starknet::class_hash::Felt252TryIntoClassHash;
+use starknet::ContractAddress;
+use starknet::contract_address_const;
+use starknet::testing;
+use traits::TryInto;
+
+use openzeppelin::tests::mocks::upgrades_v1::UpgradesV1;
+use openzeppelin::tests::mocks::upgrades_v1::IUpgradesV1Dispatcher;
+use openzeppelin::tests::mocks::upgrades_v1::IUpgradesV1DispatcherTrait;
+use openzeppelin::tests::mocks::upgrades_v2::UpgradesV2;
+use openzeppelin::tests::mocks::upgrades_v2::IUpgradesV2Dispatcher;
+use openzeppelin::tests::mocks::upgrades_v2::IUpgradesV2DispatcherTrait;
+use openzeppelin::upgrades::upgradeable::Upgradeable;
+use openzeppelin::upgrades::upgradeable::Upgradeable::Upgraded;
+use openzeppelin::tests::utils;
+
+const VALUE: felt252 = 123;
+const VALUE2: felt252 = 789;
+const SET_VALUE_SELECTOR: felt252 =
+    0x3d7905601c217734671143d457f0db37f7f8883112abd34b92c4abfeafde0c3;
+const SET_VALUE2_SELECTOR: felt252 =
+    0x47c8c185eb97d3925831a1c97e43bd9077181d2b200133ede551f1c47056a3;
+const REMOVE_SELECTOR: felt252 = 0x2beeaa48bce210364c7d2f2fbb677e08136cb29e8972a6728249364dde19e6f;
+
+fn V2_CLASS_HASH() -> ClassHash {
+    UpgradesV2::TEST_CLASS_HASH.try_into().unwrap()
+}
+
+fn CLASS_HASH_ZERO() -> ClassHash {
+    class_hash_const::<0>()
+}
+
+fn ZERO() -> ContractAddress {
+    contract_address_const::<0>()
+}
+
+//
+// Setup
+//
+
+fn deploy_v1() -> IUpgradesV1Dispatcher {
+    let calldata = array![];
+    let address = utils::deploy(UpgradesV1::TEST_CLASS_HASH, calldata);
+    IUpgradesV1Dispatcher { contract_address: address }
+}
+
+//
+// upgrade
+//
+
+#[test]
+#[available_gas(2000000)]
+#[should_panic(expected: ('Class hash cannot be zero', 'ENTRYPOINT_FAILED', ))]
+fn test_upgrade_with_class_hash_zero() {
+    let v1 = deploy_v1();
+    v1.upgrade(CLASS_HASH_ZERO());
+}
+
+#[test]
+#[available_gas(2000000)]
+fn test_persisting_selector_after_upgrade() {
+    let v1 = deploy_v1();
+    v1.set_value(VALUE);
+    assert(v1.get_value() == VALUE, 'Should be VALUE');
+
+    // Upgrade and check event
+    testing::pop_log_raw(ZERO());
```

Comment:
```suggestion
```

If there are not previously emitted events we want to remove from the queue, we don't need to call this.